src/App.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\GoodDataWriter;
+
+use Keboola\GoodData\Client;
+use Keboola\GoodData\Exception;
+use Keboola\Temp\Temp;
+use Psr\Log\LoggerInterface;
+
+class App
+{
+    /** @var  LoggerInterface */
+    protected $logger;
+    /** @var  Temp */
+    protected $temp;
+    /** @var  Client */
+    protected $gdClient;
+    /** @var ProvisioningClient  */
+    protected $gdProvisioning;
+
+    public function __construct(
+        LoggerInterface $logger,
+        Temp $temp,
+        Client $gdClient,
+        ProvisioningClient $gdProvisioning
+    ) {
+        $this->logger = $logger;
+        $this->temp = $temp;
+        $this->gdClient = $gdClient;
+        $this->gdProvisioning = $gdProvisioning;
+    }
+
+    public function checkProjectAccess(Config $config): bool
+    {
+        try {
+            $this->gdClient->get("/gdc/projects/{$config->getProjectPid()}");
+        } catch (Exception $e) {
+            if ($e->getCode() !== 403) {
+                throw $e;
+            }
+
+            if (!getenv('KBC_TOKEN')) {
+                throw new \Exception('KBC Token is missing from the environment');
+            }
+            $this->gdProvisioning->addUserToProject($config->getUserLogin(), $config->getProjectPid());
+            $this->logger->debug('GoodData user added to the project using GoodData Provisioning');
```

Comment:
sorry ze takto pidim ale mi pride ta hlaska nedostatocna - konkretne neni povedane aky user resp kvoli comu tam bol pridany. Viem si predstavit ze ked to teraz pustim a uvidim takuto hlasku tak prve co ma napadne je ze som tam bol pridany ja pod mojim mailom..Bolo by dobre zdoraznit ze ten user je na loadovanie dat gd writerom alebo ze je to user z konfigu ktory bol spusteny a kedze doteraz nebol v projektu tak sa tam pridal + jeho mail. Co myslis?